File: src/main/resources/archetype-resources/src/main/java/domain/dto/UserDTO.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.domain.dto;

import javax.validation.constraints.NotEmpty;

import ${package}.domain.generic.AuditDTO;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Vittorio Valent
 *
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class UserDTO extends AuditDTO {

	private Long id;

	@NotEmpty(message = "{UserDTO.username.NotEmpty}")
	private String username;

	@NotEmpty(message = "{UserDTO.password.NotEmpty}")
	private String password;

	private RoleDTO role;

	private Boolean accountNonExpired = Boolean.TRUE;

	private Boolean credentialsNonExpired = Boolean.TRUE;

	private Boolean accountNonLocked = Boolean.TRUE;

	private Boolean enabled = Boolean.TRUE;

	private String name;

	private String surname;

}

File: src/main/resources/archetype-resources/frontend/src/dto/user-dto.ts
import { RoleDTO } from "./role-dto";

/**
 * Classe DTO di User. DEVE essere uguale (stesso nome classe, stessi attributi e stessi nomi) a
 * quello nel backend.
 *
 *
 * @author Vittorio Valent
 */
export class UserDTO {
  id: number;

  username: string;

  password: string;

  role: RoleDTO;

  name: string;

  surname: string;

  accountNonExpired: boolean;

  credentialsNonExpired: boolean;

  accountNonLocked: boolean;

  enabled: boolean;
}
